iceoryx_utils/test/moduletests/test_index_queue.cpp

```diff
@@ -220,4 +220,82 @@ TYPED_TEST(IndexQueueTest, popIfFullReturnsNothingWhenQueueIsNotFull)
     EXPECT_FALSE(q.popIfFull().has_value());
 }
 
+TYPED_TEST(IndexQueueTest, popIfSizeIsAtLeastReturnsNothingIfQueueIsEmpty)
+{
+    auto& q = this->queue;
+    EXPECT_FALSE(q.popIfSizeIsAtLeast(1).has_value());
+}
+
+TYPED_TEST(IndexQueueTest, popIfSizeIsAtLeastZeroReturnsNothingIfQueueIsFull)
+{
+    auto& q = this->fullQueue;
+    EXPECT_FALSE(q.popIfSizeIsAtLeast(0).has_value());
+}
+
+TYPED_TEST(IndexQueueTest, popIfSizeIsAtLeastZeroReturnsNothingIfQueueIsEmpty)
+{
+    auto& q = this->queue;
+    EXPECT_FALSE(q.popIfSizeIsAtLeast(0).has_value());
+}
+
+TYPED_TEST(IndexQueueTest, popIfSizeIsAtLeastZeroReturnsNothingIfQueueContainsOneElement)
+{
+    auto& q = this->queue;
+    auto index = this->fullQueue.pop();
+    ASSERT_TRUE(index.has_value());
+    q.push(*index);
+
+    index = q.popIfSizeIsAtLeast(0u);
+    ASSERT_FALSE(index.has_value());
```

Comment:
This strange behavior can be avoided and I changed it. There is no good reason for it unless I missed something. Changed behavior and test, now it will return an index if there is one (unless capacity is 0, of course).